### PR TITLE
UrlForm and AWS query support 

### PR DESCRIPTION
**Pull request by dhpiggott**

### CODE REVIEW

File: modules/aws-http4s/src/smithy4s/aws/AwsCredentialsProvider.scala [hunk 1]
```diff
@@ -62,6 +62,8 @@ class AwsCredentialsProvider[F[_]](implicit F: Temporal[F]) {
     Resource
       .eval(fromEnv)
       .map(F.pure)
+      // TODO: Ensure minimal delay when these endpoints don't exist, e.g.
+      // when running locally.
```
Comment (kubukoz) [inline modules/aws-http4s/src/smithy4s/aws/AwsCredentialsProvider.scala:66]: related: https://github.com/disneystreaming/smithy4s/issues/1075

**Comment by dhpiggott** [comment]
Closing this because it includes unfinished support for encoding and decoding - whereas the plan is to get encoding support good enough to merge and then follow up with decoding support. So I'll open a new PR for encoding support and then a follow up will add decoding support.

**Comment by dhpiggott** [comment]
Encoding support draft: https://github.com/disneystreaming/smithy4s/pull/1112

**Comment by dhpiggott** [comment]
Decoding support will be added back after that in https://github.com/disneystreaming/smithy4s/pull/1113.

**Pull request closed without merging**